src/Neo.IO/MessageRelayer.cs

```diff
@@ -0,0 +1,197 @@
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// MessageRelayer.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Neo.IO
+{
+    public class MessageRelayer : IDisposable
+    {
+        private volatile bool _disposed;
+
+        private readonly Task[] _workers;
+        private readonly
+#if NET9_0_OR_GREATER
+            Lock
+#else
+            object
+#endif
+            _lock = new();
```

Comment:
Don't write it like this please 🥺 

```suggestion
#if NET9_0_OR_GREATER
        private readonly Lock _lock = new();
#else
        private readonly object _lock = new();
#endif
```